esmvaltool/cmorizers/obs/cmorize_obs_woa.py

```diff
@@ -4,67 +4,127 @@
-        elif var == 'o2':
-            cube *= 44.661 / 1000.  # Convert from ml/l to mol/m^3
-    return cube
 
+    if version == '2018':
+        with constant_metadata(cube):
+            if var in ['o2', 'po4', 'si', 'no3']:
+                cube /= 1000.  # Convert from umol/kg to mol/m^3
+
+    if version == '2013v2':
+        with constant_metadata(cube):
+            mll_to_mol = ['po4', 'si', 'no3']
+            if var in mll_to_mol:
+                cube /= 1000.  # Convert from ml/l to mol/m^3
+            elif var == 'thetao':
+                cube += 273.15  # Convert to Kelvin
+            elif var == 'o2':
+                cube *= 44.661 / 1000.  # Convert from ml/l to mol/m^3
+
+    return cube
 
-def extract_variable(var_info, raw_info, out_dir, attrs, year):
-    """Extract to all vars."""
-    var = var_info.short_name
-    cubes = iris.load(raw_info['file'])
-    rawvar = raw_info['name']
 
-    for cube in cubes:
-        if cube.var_name == rawvar:
-            fix_var_metadata(cube, var_info)
-            convert_timeunits(cube, year)
-            fix_coords(cube)
-            _fix_data(cube, var)
-            set_global_atts(cube, attrs)
-            save_variable(
-                cube, var, out_dir, attrs, unlimited_dimensions=['time'])
+def collect_files(in_dir, var, cfg):
+    """Compose input file list and download if missing."""
+    file_list = []
+    var_dict = cfg['variables'][var]
+    in_dir = os.path.join(in_dir, var_dict['name'])
+
+    fname = cfg['attributes']['short_name'].lower(
+    ) + '_' + var_dict['file'] + '00_01.nc'
+    in_file = os.path.join(in_dir, fname)
+
+    # download if missing
+    if not os.path.isfile(in_file):
+        if not os.path.isdir(in_dir):
+            os.makedirs(in_dir)
+        logger.info('Input file %s is missing. Start download ... ', fname)
+        url = os.path.join(cfg['attributes']['source'], var_dict['name'],
+                           'netcdf', var_dict['file'].split('_')[0],
+                           cfg['custom']['resolution'], fname)
+        url_file = requests.get(url)
+        open(in_file, 'wb').write(url_file.content)
+
+    file_list.append(in_file)
+
+    return file_list
+
+
+def extract_variable(in_files, out_dir, attrs, raw_info, cmor_table):
+    """Extract variables and create OBS dataset."""
+    var = raw_info['var']
+    var_info = cmor_table.get_variable(raw_info['mip'], var)
+    rawvar = raw_info['raw_var']
+    with catch_warnings():
+        filterwarnings(
+            action='ignore',
+            message='Ignoring netCDF variable .* invalid units .*',
+            category=UserWarning,
+            module='iris',
+        )
+        cubes = iris.load(in_files, rawvar)
+    iris.util.equalise_attributes(cubes)
+    cube = cubes.concatenate_cube()
+
+    # set reference time
+    year = raw_info['reference_year']
+    cube.coord('time').climatological = False
+    cube.coord('time').points = 6.5
```

Comment:
Why 6.5 and why does this need to be hardcoded?

Reply:
Time in the original files is rather messy and set manually the center value to 6.5 expressed as month of the year.